include/librealsense2/rs.hpp

```diff
@@ -30,6 +30,107 @@ namespace rs2
         error::handle(e);
     }
 
+    /*
+        Interface to the log message data we expose.
+    */
+    class log_message
+    {
+        // Only log_callback should be creating us!
+        template< class T > friend class log_callback;
+
+        log_message( rs2_log_message const & msg ) : _msg( msg ) {}
+
+    public:
+        /* 
+            Builds a complete log message, as defined by librealsense, with level, timestamp, etc.:
+                11/12 13:49:40,153 INFO [10604] (rs.cpp:2271) Framebuffer size changed to 1552 x 919
+        */
+        std::shared_ptr< rs2_string > build() const
```

Comment:
Can the combined string be pre-computed within the log struct, and exposed as char* as well?

Reply:
Yes, but then you're assuming how the client wants to format the result. Usually you want the raw message as it was written, and leave the formatting to the sink implementer.